dts: arm: nrf54l: Add nodes required by MCUboot

michalek-no opened the pull request:
The commit adds nodes needed by MCUboot.
It also modifies chosen list by addition of MCUmgr port and code-partition.

michalek-no committed:
dts: arm: nrf54l: Add nodes required by MCUboot

The commit adds partitons on rram0.

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>

nvlsianpu reviewed (commented):
Only fixed-partitions are modified right now in this patch. Can you redact the commit message?

nordicjm commented on boards/arm/nrf54l15pdk_nrf54l15/nrf54l15pdk_nrf54l15_cpuapp.dts:
```diff
@@ -133,13 +133,30 @@
-		storage_partition: partition@f2000 {
+		slot0_partition: partition@10000 {
+			label = "image-0";
+			reg = <0x10000 DT_SIZE_K(349)>;
+		};
+		slot0_ns_partition: partition@67400 {
+			label = "image-0-nonsecure";
+			reg = <0x67400 DT_SIZE_K(349)>;
+		};
+		slot1_partition: partition@be800 {
+			label = "image-1";
+			reg = <0xbe800 DT_SIZE_K(349)>;
+		};
+		slot1_ns_partition: partition@115c00 {
+			label = "image-1-nonsecure";
+			reg = <0x115c00 DT_SIZE_K(349)>;
```
Hate to be a pain but can we make this divisable by 4 so that external memory devices e.g. mx25 can be used, otherwise it is impossible? 348KiB for each of these then add 4KiB to settings to make that 36KiB

fabiobaltieri commented:
@anangl ping, do you folks want to merge this for 3.6? It's not quite a bugfix bug you may make an argument about it

fabiobaltieri commented:
> @anangl ping, do you folks want to merge this for 3.6? It's not quite a bugfix bug you may make an argument about it

@anangl ^^

anangl commented:
I don't know if this is needed for 3.6. @michalek-no @nvlsianpu Is it?

nvlsianpu commented:
@anangl We need this for NCS 2.6.0
Don't know whether this imply that it should go in v3.6

nvlsianpu committed:
linker: Set ROM offset for MCUboot with NRF54L

Due to number of IRQs the ROM offset, that contains MCUboot,
needs to be moved to address that is aligned with 2048 byte
long vector table.
This comes from Cortex-M requirement where vector table needs
to be placed at offset equal to lowest power of two number
that will fit vector table.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>
Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>
Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

michalek-no committed:
dts: arm: add uart-mcumgr choosen for 54L15

setting uart20 as default uart-mcumgr

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>

nvlsianpu commented on Kconfig.zephyr:
```diff
@@ -129,6 +129,7 @@ endif # HAS_FLASH_LOAD_OFFSET
 config ROM_START_OFFSET
 	hex
 	prompt "ROM start offset" if !BOOTLOADER_MCUBOOT
+	default 0x800 if BOOTLOADER_MCUBOOT && SOC_PLATFORM_NRF54L
```
might be moved to Kconfig.defconfig

michalek-no commented:
rebase

nordicjm commented on boards/nordic/nrf54l15pdk/Kconfig.defconfig:
```diff
@@ -6,4 +6,8 @@ if BOARD_NRF54L15PDK_NRF54L15_CPUAPP
 	default BT
 
+config ROM_START_OFFSET
+	hex
```
```suggestion
```

gchwier reviewed (approved):
Tested with PR https://github.com/zephyrproject-rtos/zephyr/pull/70105 (removed CONFIG_FPROTECT).
Building and flashing with enabled MCUboot works good on nrf54l15pdk,
used command:
`west build --sysbuild -p -b nrf54l15pdk/nrf54l15/cpuapp ${ZEPHYR_BASE}/samples/hello_world -d build-54l-hello -- \
-DSB_CONFIG_BOOTLOADER_MCUBOOT=y`

I was able also to run tests with `mcumgr`:
`${ZEPHYR_BASE}/scripts/twister -vv -T ${ZEPHYR_BASE}/tests/boot/with_mcumgr --west-flash --device-testing --device-serial /dev/ttyACM1 --force-platform -p nrf54l15pdk/nrf54l15/cpuapp --enable-slow -c`
Following test scenarios were successfully executed:
  _boot.with_mcumgr.test_upgrade.test_upgrade_with_confirm
  boot.with_mcumgr.test_upgrade.test_upgrade_with_revert
  boot.with_mcumgr.test_upgrade.test_upgrade_signature[no_key]
  boot.with_mcumgr.test_upgrade.test_upgrade_signature[invalid_key]
  boot.with_mcumgr.test_downgrade_prevention.test_downgrade_prevention_

nvlsianpu commented:
@carlescufi Can this be merged?

fabiobaltieri commented:
> @carlescufi Can this be merged?

Formally this needs @anangl approva, we have a dashboard and this does not even show as mergeable until the approver approved. Or we can change the approver, does not touch any core file so I suppose it's fine to move over to one of the mcubook collabs.

fabiobaltieri commented:
That was easy, thanks @anangl :-)
